nightmare/paths/graphs.spec.js

```diff
@@ -0,0 +1,127 @@
+  afterAll(async () => nightmare.end());
+
+  describe('login', () => {
+    it('should log in as admin then click the settings nav button', done => {
+      loginAsAdminAndGoToSettings(nightmare, catchErrors, done);
+    });
+  });
+
+  describe('Graphs in Page', () => {
+    it('should create a basic page', async () => {
+      await nightmare
+        .goto('http://localhost:3000/en/settings')
+        .clickLink('Pages')
+        .clickLink('Add page')
+        .wait('.page-creator')
+        .write(localSelectors.pageTitleInput, 'Page data viz')
+        .write(localSelectors.pageContentsInput, '</p><Dataset />')
+        .clickLink('Save')
+        .wait(1000);
+
+      await nightmare.getInnerText(localSelectors.createdPageLink).then(text => {
+        expect(text).toContain('(view page)');
+      });
+    });
+
+    it('should insert Bar chart graph in created page', async () => {
+      await nightmare
+        .evaluate(
+          selector => document.querySelector(selector).value,
+          localSelectors.pageContentsInput
+        )
+        .then(response => {
+          expect(response).toBeTruthy();
+          expect(response).toContain('<Dataset />');
+        });
+      await nightmare
+        .write(localSelectors.pageContentsInput, graphs.barChart)
+        .click(localSelectors.savePageButton);
+    });
+
+    it('should display Bar chart graph in page with no more than a 1% difference', async () => {
+      await nightmare
+        .evaluate(selector => document.querySelector(selector).href, localSelectors.createdPageLink)
+        .then(link => nightmare.goto(link));
+
+      await nightmare
+        .wait('div.markdown-viewer')
+        .getInnerHtml('div.markdown-viewer')
+        .then(html => {
+          expect(html).toContain('class="recharts-responsive-container"');
+
+          return nightmare
+            .waitForGraphsAnimation()
+            .screenshot()
+            .then(image => {
+              expect(image).toMatchImageSnapshot({
+                failureThreshold: 0.01,
+                failureThresholdType: 'percent',
+                allowSizeMismatch: true,
+              });
+            });
+        });
+    });
+
+    it('should navigate back to the edit page and insert Pie+List chart graphs', async () => {
+      await nightmare
+        .back()
+        .wait('.page-creator')
+        .clearInput(localSelectors.pageContentsInput)
+        .write(localSelectors.pageContentsInput, '</p><Dataset />')
+        .write(localSelectors.pageContentsInput, graphs.pieChart)
+        .write(localSelectors.pageContentsInput, graphs.listChart)
+        .click(localSelectors.savePageButton);
+    });
+
+    it('should display Pie-List chart graphs in page with no more than a 1% difference', async () => {
+      await nightmare
+        .evaluate(selector => document.querySelector(selector).href, localSelectors.createdPageLink)
+        .then(link => nightmare.goto(link));
```

Comment:
@eric we discussed about this, i do not remember why this is not just a clickLink("Label") ??

Reply:
@joan the initial issue with clickLink was that it was opening a new window on links with the `target="_blank" attribute and Nightmare have no way yet to switch focus from the old to the new window. Some of the scope of that issue is documented here https://github.com/segmentio/nightmare/issues/804. I however, modified the `clickLink` action to remove the target attribute of an anchor element to avoid opening the link in a new window